src/core/tensor/tensor_math_cuda.h

```diff
@@ -20,260 +20,428 @@
+  float* outPtr = static_cast<float*>(out->block()->mutable_data());
+  const size_t num = in->Size();
   cuda::mult(num, inPtr, x, outPtr, ctx->stream);
 }
 /// out = in1 * in2
 template <>
-void EltwiseMult<float, lang::Cuda>(const size_t num, const Block* in1,
-                                    const Block* in2, Block* out,
+void EltwiseMult<float, lang::Cuda>(const Tensor* in1,
+                                    const Tensor* in2, Tensor* out,
                                     Context* ctx) {
-  const float* inPtr1 = static_cast<const float*>(in1->data());
-  const float* inPtr2 = static_cast<const float*>(in2->data());
-  float* outPtr = static_cast<float*>(out->mutable_data());
+  const float* inPtr1 = static_cast<const float*>(in1->block()->data());
+  const float* inPtr2 = static_cast<const float*>(in2->block()->data());
+  float* outPtr = static_cast<float*>(out->block()->mutable_data());
+  const size_t num = in1->Size();
   cuda::mult(num, inPtr1, inPtr2, outPtr, ctx->stream);
 }
 /// Base is e. out[i]=e^in[i]
 template <>
-void Exp<float, lang::Cuda>(const size_t num, const Block* in, Block* out,
+void Exp<float, lang::Cuda>(const Tensor* in, Tensor* out,
                             Context* ctx) {
-  const float* inPtr = static_cast<const float*>(in->data());
-  float* outPtr = static_cast<float*>(out->mutable_data());
+  const float* inPtr = static_cast<const float*>(in->block()->data());
+  float* outPtr = static_cast<float*>(out->block()->mutable_data());
+  const size_t num = in->Size();
   cuda::exp(num, inPtr, outPtr, ctx->stream);
 }
 
 template <>
-void GE<float, lang::Cuda>(const size_t num, const Block* in, const float x,
-                           Block* out, Context* ctx) {
-  float* outPtr = static_cast<float*>(out->mutable_data());
-  const float* inPtr = static_cast<const float*>(in->data());
+void GE<float, lang::Cuda>(const Tensor* in, const float x,
+                           Tensor* out, Context* ctx) {
+  float* outPtr = static_cast<float*>(out->block()->mutable_data());
+  const float* inPtr = static_cast<const float*>(in->block()->data());
+  const size_t num = in->Size();
   cuda::ge(num, inPtr, x, outPtr, ctx->stream);
 }
 template <>
-void GE<float, lang::Cuda>(const size_t num, const Block* in1, const Block* in2,
-                           Block* out, Context* ctx) {
-  float* outPtr = static_cast<float*>(out->mutable_data());
-  const float* inPtr1 = static_cast<const float*>(in1->data());
-  const float* inPtr2 = static_cast<const float*>(in2->data());
+void GE<float, lang::Cuda>(const Tensor* in1, const Tensor* in2,
+                           Tensor* out, Context* ctx) {
+  float* outPtr = static_cast<float*>(out->block()->mutable_data());
+  const float* inPtr1 = static_cast<const float*>(in1->block()->data());
+  const float* inPtr2 = static_cast<const float*>(in2->block()->data());
+  const size_t num = in1->Size();
   cuda::ge(num, inPtr1, inPtr2, outPtr, ctx->stream);
```

Comment:
in1 and in2 may have different strides.

Reply:
one approach to avoid handling the strides:
C=Sub(A, B) (C has the same shape/stride as A or B). Sub is implemented using cudnnOpTensor.
then call cuda::ge(C, 0), which compares C[i] with 0 and set C[i]=1.0 if positive or 0; It does not need to consider the stride of C.